### PR TITLE
Command line option to convert and upload training forms

### PR DESCRIPTION
**Pull request by latin-panda**
# Description

- Adds `convert-training-forms`, `upload-training-forms` and  `validate-training-forms` commands for training forms

https://github.com/medic/cht-core/issues/6598

[CHT-Core PR](https://github.com/medic/cht-core/pull/8017) - Training cards feature
[CHT-Docs PR](https://github.com/medic/cht-docs/pull/970) - Technical guide

# Code review items

- Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- Tested: Unit and/or integration tests where appropriate
- Backwards compatible: Works with existing data and configuration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: src/fn/watch-project.js [hunk 1]
```diff
@@ -125,6 +130,44 @@ const processAppFormMedia = (formMediaDir, fileName) => {
     return false;
 };
 
+const processTrainingForm = (eventType, fileName) => {
```
Comment (dianabarsan) [inline src/fn/watch-project.js:133]: I'm wondering if we can reduce some of this code duplication. 
It looks like the only difference between `processTrainingForm` and `processAppForm` are the `validate???Forms`, `upload???Forms` and `upload???Forms`. Is it possible we could converge this code into a single function that knows which functions to call based on the type of form it is?

**Review by dianabarsan — COMMENTED** [review]
Cool stuff! Thanks for adding this! 
I left one refactoring suggestion inline.

File: src/fn/watch-project.js (hunk 1 above)
Comment (latin-panda) [reply src/fn/watch-project.js]: I've made a refactor

**Comment by latin-panda** [comment]
Hi @dianabarsan I've resolved the feedback and pushed a refactor, when you have time, can you please have a look? Thanks!

File: test/fn/watch-project.spec.js [hunk 2]
```diff
@@ -273,6 +275,55 @@ describe('watch-project', function () {
...
+  it('watch-project: convert training forms', () => {
+    const form = 'new_actions_training';
+    const copyForm = () => copySampleForms('training-xlsx', TRAINING_FORMS_PATH);
+
+    return watchWrapper(copyForm, `${form}.xlsx`)
+      .then(() => {
+        const forms = fs.fs.readdirSync(trainingFormDir);
+        expect(forms).to.include(`${form}.xml`);
+      })
+      .then(() => cleanFormDir(trainingFormDir, form));
```
Comment (dianabarsan) [inline test/fn/watch-project.spec.js:287]: Is it possible to move `cleanFormDir` in an `afterEach` hook? I would expect that each test should start with an empty form directory.

**Review by dianabarsan — APPROVED** [review]
Very nice. I left one minor improvement request around test cleanup.

**Comment by latin-panda** [comment]
@dianabarsan Please hold the review, I've clicked by mistake the review icon.

File: test/fn/watch-project.spec.js (hunk 2 above)
Comment (latin-panda) [reply test/fn/watch-project.spec.js]: Done

**Comment by latin-panda** [comment]
@dianabarsan thanks for the review, I've applied your feedback and rebased my branch. I'm going to send this piece to AT with a test recommendation.

**Review by dianabarsan — APPROVED** [review]
Nice stuff you have here :D

**Pull request merged**